### ISSUE TITLE
GitVersionTaskAssemblyInfo.g.cs Conflict with Project Namespace

### ISSUE THREAD

**Issue by sponge-b0b** [GitTools/GitVersion#709]
My project has a namespace like Foo.System. Since you use the fully qualified namespace of the attribute [System.Runtime.CompilerServices.CompilerGenerated] for the GitVersionInformation class and since it is nested within the Foo.System namespace of my project it won't compile because the compiler is looking for a Runtime namespace within Foo.System which doesn't exist. The solution is to add a using System.Runtime.CompilerServices; line and adorn the class with just [CompilerGenerated] like so:
~~~
using System;
using System.Reflection;
using System.Runtime.CompilerServices;  // Add this line

[assembly: AssemblyVersion(""{0}"")]
[assembly: AssemblyFileVersion(""{1}"")]
[assembly: AssemblyInformationalVersion(""{2}"")]

namespace Foo.System
{
    // Modify this line
    // Compiler sees this line as [Foo.System.Runtime.CompilerServices.CompilerGenerated], there is no Runtime in Foo.System
    [System.Runtime.CompilerServices.CompilerGenerated]
    
    // To this
    [CompilerGenerated]
    static class GitVersionInformation
~~~
Now there is no more conflict. Ideally it would be nice if the auto-generation of GitVersionInformation was configurable and not hard-coded in AssemblyInfoBuilder.cs. Or if you provided a template file for GitVersionTaskAssemblyInfo.g.cs that could be customized.

Thank you for your help.

**Comment by JakeGinnivan** [comment]
Hi @sponge-b0b 

Happy to take a pull request to fix this issue. I suggest changing it to `[global::System.Runtime.CompilerServices.CompilerGenerated]` to force it to reference the global namespace.

As for the ability to configure, maybe open up another issues just for that and put some ideas on how you think that should work? Would we force template files for all languages (f#, vb.net and c#)?

**Comment by sponge-b0b** [comment]
Hi @JakeGinnivan

Thanks for your help. I will create a pull request and think about how the configure could work.

**Comment by gep13** [comment]
@sponge-b0b did you mean to close this issue?  There is no harm in keeping it open, and referencing it from your PR.

**Comment by sponge-b0b** [comment]
@gep13 oops, fat-fingered it. Thanks.

**Comment by gep13** [comment]
:+1:

**Comment by JakeGinnivan** [comment]
Fixed by #710
